### PR TITLE
Add timeout check

### PR DESCRIPTION
**Pull request by vctt94**
This PR adds a timeout when selecting new VSPs.

### CODE REVIEW

**Comment by alexlyp** [comment]
Not quite fixing the problem that I've seen.  Currently I've seen it time out during getVSPInfo action.  Should be easily reproducible with test.stakey.com no longer responding.

**Comment by vctt94** [comment]
Here it works as expected

image

You select the vsp, wait 5 seconds, and nothing happens?

**Comment by alexlyp** [comment]
Right I'm talking about during ProcessManaged/Unmanaged during launch. Hangs for about a minute waiting for a response then errors out.

File: app/actions/VSPActions.js [hunk 1]
```diff
@@ -14,20 +14,29 @@ export const GETVSP_ATTEMPT = "GETVSP_ATTEMPT";
...
-    dispatch({ type: GETVSP_SUCCESS, info });
-    return info.data;
-  } catch (error) {
-    dispatch({ type: GETVSP_FAILED, error });
-    return { error };
-  }
-};
+  // 5 seconds for timeout.
+  const TIMEOUT_TIME = 5000;
+  // add https into vsp host, so we can fetch its information.
```
Comment (amassarwi) [inline app/actions/VSPActions.js:21]: string template

File: app/actions/VSPActions.js [hunk 2]
```diff
@@ -14,20 +14,29 @@ export const GETVSP_ATTEMPT = "GETVSP_ATTEMPT";
...
+  host = "https://" + host;
+  wallet.allowVSPHost(host);
+  const timeout = new Promise((resolve, reject) => setTimeout(() => {
```
Comment (amassarwi) [inline app/actions/VSPActions.js:24]:
```suggestion
  const timeout = new Promise((resolve, reject) => setTimeout(() => 
```

**Review by amassarwi — APPROVED** [review]
LGTM, except two minors.

File: app/actions/VSPActions.js (hunk 1 above)
Comment (vctt94) [reply app/actions/VSPActions.js:21]: not sure what you mean here. Move this to constants file?

Comment (amassarwi) [reply app/actions/VSPActions.js:21]: ops was wrong line, see line 22 suggestion.

File: app/actions/VSPActions.js [hunk 3]
```diff
@@ -14,20 +14,29 @@ export const GETVSP_ATTEMPT = "GETVSP_ATTEMPT";
...
-    return info.data;
-  } catch (error) {
-    dispatch({ type: GETVSP_FAILED, error });
-    return { error };
-  }
-};
+  // 5 seconds for timeout.
+  const TIMEOUT_TIME = 5000;
+  // add https into vsp host, so we can fetch its information.
+  host = "https://" + host;
```
Comment (amassarwi) [inline app/actions/VSPActions.js:22]:
```suggestion
  host = `https://${host}`;
```

**Review by alexlyp — APPROVED** [review]
tACK

**Pull request merged**